typescript/core/src/v0/ChunkCursor.ts

```diff
@@ -292,3 +291,28 @@ export class ChunkCursor {
     return cursor.records[cursor.index]![0];
   }
 }
+
+// Binary search based on lodash's sortedIndexBy()
+function sortedIndexBy<T>(array: T[], value: T, iteratee: (value: T) => number | bigint): number {
+  let low = 0;
+  let high = array == null ? 0 : array.length;
+  if (high === 0) {
+    return 0;
+  }
+
+  const valueNumber = iteratee(value);
+  const valIsNaN = valueNumber !== valueNumber;
+
+  while (low < high) {
+    const mid = Math.floor((low + high) / 2);
```

Comment:
You might want to read about the adjustment to the math made in: https://www.npmjs.com/package/binary-search